types/id.go

```diff
@@ -1,12 +1,13 @@
 package types
 
-import sdk "github.com/cosmos/cosmos-sdk/types"
-
-type ModuleID struct {
-	ModuleName string
-	Path       []byte
-}
-
-func (m ModuleID) Address() sdk.AccAddress {
-	return AddressHash(m.ModuleName, m.Path)
+// ModuleAcc represents an account managed by a module
+type ModuleAcc struct {
+	// Module name managing this account. Used by router, used when registering a MsgServer
+	// and invoking service methods.
+	Module string
+	// Key for identifying cross-module messages and module address derivation
+	// TODO: probably we don't need it?
+	Key []byte
+	// Address derived from other module account
+	Address []byte
```

Comment:
Why is this public? Isn't it just a cache of the computed address? In which case it could be private with an accessor method which constructs it if needed?

Reply:
In fact, we don't need `Key` - it's never used. ModuleAcc is used only for the  router by a module key. And the module key copies the `Key` and sets the `Address`. So instead - we can remove `Key` and only use `Address`.

Reply:
It's used for derived module keys. Didn't we talk about that?